packages/react-dom/src/client/setAttribute.js

```diff
@@ -16,9 +18,17 @@ import type {TrustedValue} from './ToStringValue';
 export function setAttribute(
   node: Element,
   attributeName: string,
-  attributeValue: string | TrustedValue,
+  attributeValue: string | TrustedValue | IdObject,
 ) {
-  node.setAttribute(attributeName, (attributeValue: any));
+  if (
+    typeof attributeValue === 'object' &&
```

Comment:
This is probably React’s hottest path that we’re making slower (and not inlinable). Is this the best place we can do this? Where does TrustedValue do its check?